### ISSUE TITLE
Assignable judging rules is not consistent between variable and function

### ISSUE THREAD

**Issue by k8w** [microsoft/TypeScript#43945]
# Bug Report

### 🔎 Search Terms
`assignable rule` `function assignable extend`

### 🕗 Version & Regression Information
4.2.4

- This is the behavior in every version I tried, and I reviewed the FAQ for entries.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.3.0-beta#code/JYOwLgpgTgZghgYwgAgCoQM5mQbwLABQyxyAbnADYCuEAXMllKAOaEC+hhYAngA4rosARmQBeNJmwAyXMgAeC+oxbI2Abk4EKEbHACE9QWA1adyAEYGJwk3DEWT5+3BOFtuoVYAUpQ5ICUYgB8ZAD2wAAmJu4WnvQ+fsKBoiGk4VGEcCLi5kKO2chZakA)

### 💻 Code

```ts
interface Test {
    value: string
}

type Test1 = Test & { xxx: string };

let a!: Test;
let b!: Test1;
// No error (work as intended)
a = b;
// Got error here (work as intended)
b = a;

let a1!: (v: Test) => void;
let b1!: (v: Test1) => void;
// UNEXPECTED: Got error here (not consistent with above)
a1 = b1;
// UNEXPECTED: But here got no error (not consistent with above)
b1 = a1;
```

See `Test1` is extended from `Test`, so it should be assignable to assign `Test1` to type `Test`, this works as intended.
But when it switch to function, the rules changed to the opposite.

### 🙁 Actual behavior
`(v: Test1)=>void` is NOT assignable to `(v: Test)=>void`

### 🙂 Expected behavior
`(v: Test1)=>void` is assignable to `(v: Test)=>void`

**Comment by jcalz** [comment]
This is working as intended.  Functions need to be contravariant in their parameters in order to maintain type safety.  See the documentation for [the `--strictFunctionTypes` compiler flag](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#strict-function-types) for more details.

If `a1 = b1` worked without error, then this would happen:

```ts
let b1 = (v: Test1) => console.log(v.xxx.toUpperCase());
let a1: (v: Test) => void = b1; // error, for good reason

a1({ value: "val" }); // no compiler error, but EXPLOSION AT RUNTIME, v.xxx is undefined
```

[Playground link to code](https://www.typescriptlang.org/play?ts=4.2.3#code/JYOwLgpgTgZghgYwgAgCoQM5mQbwLABQyxyAbnADYCuEAXMllKAOaEC+hYAngA4rpYAjMgC8aTNgBkuZAA959Ri2RsA3IUIUI2AEbCxAClL0BYQQEpRAPmQIA9iAx2tAOgp3mRl-NkuwdgFUePigAYTgMCANzc3UCLWw4QXojEwlLERtSO2AAE1FkPVVkAHoS5GgoOygAGmQYauRmOzt8qAgIhw0CJIMcMkoaegAicgphlVjS8oBRAA0ABQAZAHkAZQBJFYA5ZABBVGQAJQDt1A2AWRm60m95ZGAMZCoQXIgYUAhcwiA)
